### PR TITLE
BGP af merge, with enhancements to utilitylib

### PR DESCRIPTION
**Pull request by chburket**
Much like the new unsupported_properties method I added to utilitylib, this PR contains a new dependency_manifest method that can be overridden by a test to return manifest config for any test dependencies.

I left Helen's method of passing this dependency config into test_harness_run, which I still think can still be useful on a testcase be testcase basis.

### CODE REVIEW

File: tests/beaker_tests/cisco_bgp/test_bgp.rb [hunk 1]
```diff
@@ -207,10 +207,13 @@ def unsupported_properties(tests, id)
...
       unprops <<
         :disable_policy_batching <<
+        :enforce_first_as <<
         :event_history_cli <<
         :event_history_detail <<
         :event_history_events <<
         :event_history_periodic <<
+        :fast_external_fallover <<
+        :flush_routes <<
         :neighbor_down_fib_accelerate
```
Comment (chburket) [inline tests/beaker_tests/cisco_bgp/test_bgp.rb]: These changes were to fix the bgp beaker tests under nx-os (see the note I sent to agent-coders).

I will push README changes in a few minutes.

File: tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb [hunk 2]
```diff
@@ -49,6 +49,61 @@ def unsupported_properties(_, _)
   ]
 end
```
Comment (chburket) [inline tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb]: I updated test_bgpneightboraf to use the new dependency_manifest way of specifying dependency config.  I can put it back if you (Helen) think it was better the old way.

Comment (HelenHZhang) [inline tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb]: Your change looks OK with me.

**Comment by HelenHZhang** [comment]
A general comment about "asn" number. In test_bgp_af, we define "asn" in the beginning, refer it everywhere else, in other files, we use constant everywhere. Should we make some effort in the future (when we merge it with develop branch)?
You don't have to change it right now.

**Comment by chburket** [comment]
I actually like the way we had it before this merge (BgpLib:ASN), but we can put it back that way after we merge back into develop (if we want).

File: tests/beaker_tests/cisco_bgp_af/test_bgpaf.rb [hunk 3]
```diff
@@ -643,104 +208,82 @@ def test_vpn_afis(tests, platform, desc)
   },
 }
 
-redistribute = Array[%w(direct d_map), %w(static s_map)]
+redistribute = Array[%w(static RouteMap), %w(lisp RouteMap)]
```
Comment (glennmatthews) [inline tests/beaker_tests/cisco_bgp_af/test_bgpaf.rb]: why did this change from direct, static to static, lisp?

**Comment by HelenHZhang** [comment]
:+1:

File: tests/beaker_tests/cisco_bgp_af/test_bgpaf.rb (hunk 3 above)
Comment (chburket) [inline tests/beaker_tests/cisco_bgp_af/test_bgpaf.rb]: "direct" is not supported on XR, and "lisp" is supported on both.  It didn't seem like the "direct" value was chosen for a particular reason other than it made for cleaner ruby code than the value that was there before the develop refactor ("eigrp 1").

**Comment by glennmatthews** [comment]
:+1:

**Pull request merged**